client/auth/login.jsx

```diff
@@ -67,16 +67,36 @@ module.exports = React.createClass( {
 
 	componentWillUnmount: function() {
 		AuthStore.off( 'change', this.refreshData );
+		clearTimeout( this.pollTimeout );
+		this.pollTimeout = null;
 	},
 
 	refreshData: function() {
 		this.setState( AuthStore.get() );
```

Comment:
These can be separated, and it gets much clearer this way. There was a UI action that directly touched the state, which is now more neatly extracted in the store actions (`USE_AUTH_CODE`).